code/modules/mob/living/carbon/human/species_types/ipc.dm

```diff
@@ -0,0 +1,189 @@
+	if(!screen_choice)
+		return
+	if(!color_choice)
+		return
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	H.dna.features["ipc_screen"] = screen_choice
+	H.eye_color = sanitize_hexcolor(color_choice)
+	H.update_body()
+
+/obj/item/apc_powercord
+	name = "power cord"
+	desc = "An internal power cord hooked up to a battery. Useful if you run on electricity. Not so much otherwise."
+	icon = 'icons/obj/power.dmi'
+	icon_state = "wire1"
+
+/obj/item/apc_powercord/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!istype(target, /obj/machinery/power/apc) || !ishuman(user) || !proximity_flag)
+		return ..()
+	user.changeNext_move(CLICK_CD_MELEE)
+	var/obj/machinery/power/apc/A = target
+	var/mob/living/carbon/human/H = user
+	var/obj/item/organ/stomach/cell/cell = H.internal_organs_slot["stomach"]
+	if(!cell)
+		to_chat(H, "<span class='warning'>You try to siphon energy from the [A], but your power cell is gone!</span>")
+		return
+
+	if(A.obj_flags & EMAGGED || A.stat & BROKEN)
+		do_sparks(3, FALSE, A)
+		to_chat(H, "<span class='warning'>The [A] power currents surge erratically, damaging your chassis!</span>")
+		H.adjustFireLoss(10)
+		return
+
+	if(A.cell && A.cell.charge > 0)
+		if(H.nutrition >= NUTRITION_LEVEL_WELL_FED)
+			to_chat(user, "<span class='warning'>You are already fully charged!</span>")
+			return
+		else
+			powerdraw_loop(A, H)
+			return
+
+	to_chat(user, "<span class='warning'>There is no charge to draw from that APC.</span>")
+
+/obj/item/apc_powercord/proc/powerdraw_loop(obj/machinery/power/apc/A, mob/living/carbon/human/H)
+	H.visible_message("<span class='notice'>[H] inserts a power connector into the [A].</span>", "<span class='notice'>You begin to draw power from the [A].</span>")
+	while(do_after(H, 10, target = A))
+		if(loc != H)
+			to_chat(H, "<span class='warning'>You must keep your connector out while charging!</span>")
+			break
+		if(A.cell.charge == 0)
+			to_chat(H, "<span class='warning'>The [A] doesn't have enough charge to spare.</span>")
+			break
+		A.charging = 1
+		if(A.cell.charge >= 500)
+			H.nutrition += 50
+			A.cell.charge -= 250
+			to_chat(H, "<span class='notice'>You siphon off some of the stored charge for your own use.</span>")
+		else
+			H.nutrition += A.cell.charge/10
+			A.cell.charge = 0
+			to_chat(H, "<span class='notice'>You siphon off as much as the [A] can spare.</span>")
+			break
+		if(H.nutrition > NUTRITION_LEVEL_WELL_FED)
+			to_chat(H, "<span class='notice'>You are now fully charged.</span>")
+			break
+	H.visible_message("<span class='notice'>[H] unplugs from the [A].</span>", "<span class='notice'>You unplug from the [A].</span>")
+
+/datum/species/ipc/spec_life(mob/living/carbon/human/H)
+	. = ..()
+	if(H.health <= HEALTH_THRESHOLD_CRIT && H.stat != DEAD) // So they die eventually instead of being stuck in crit limbo.
+		H.adjustFireLoss(6) // After bodypart_robotic resistance this is ~2/second
```

Comment:
even considering the fire loss multiplier?

Reply:
If I recall correctly, yes. Your mileage may vary here, however: I had to make some augmented limb adjustments because it gave IPCs ash storm immunity.